File: data/r2dbc/r2dbc-flyway-postgres/src/main/java/com/mahmud/r2dbcflywaypostgres/repository/TodoRepository.java
package com.mahmud.r2dbcflywaypostgres.repository;

import com.mahmud.r2dbcflywaypostgres.domain.Todo;
import org.springframework.data.r2dbc.repository.R2dbcRepository;
import reactor.core.publisher.Flux;

public interface TodoRepository extends R2dbcRepository<Todo, Long> {
    Flux<Todo> findByCompleted(Boolean completed);
}

File: io/grpc-mongodb-todo/grpc-server/src/main/kotlin/com/mahmud/grpcserver/repository/TodoRepository.kt
package com.mahmud.grpcserver.repository

import com.mahmud.grpcserver.model.Todo
import org.springframework.data.mongodb.repository.ReactiveMongoRepository
import reactor.core.publisher.Flux


interface TodoRepository : ReactiveMongoRepository<Todo, String> {
    fun findByCompleted(completed: Boolean): Flux<Todo>
}